src/main/java/net/openhft/chronicle/queue/reader/ContentBasedLimiter.java

```diff
@@ -0,0 +1,10 @@
+package net.openhft.chronicle.queue.reader;
+
+import net.openhft.chronicle.wire.DocumentContext;
+
+public interface ContentBasedLimiter {
+
+    boolean shouldHaltReading(DocumentContext documentContext);
```

Comment:
This is kind-of a predicate, but with a proper name, it's clearer what the return value means.